bliss/datasets/galsim_galaxies.py

```diff
@@ -163,24 +213,36 @@ def _draw_pareto_flux(self, n_samples=1) -> Tensor:
         uniform_samples = torch.rand(n_samples) * u_max
         return self.min_flux / (1.0 - uniform_samples) ** (1 / self.alpha)
 
-    def draw_galaxy_params(self, n_samples=1):
-        # create galaxy as mixture of Exponential + DeVacauleurs
-        if self.flux_sample == "uniform":
-            total_flux = self._uniform(self.min_flux, self.max_flux, n_samples=n_samples)
-        elif self.flux_sample == "pareto":
-            total_flux = self._draw_pareto_flux(n_samples=n_samples)
-        else:
-            raise NotImplementedError()
-        disk_frac = self._uniform(0, 1, n_samples=n_samples)
-        beta_radians = self._uniform(0, 2 * np.pi, n_samples=n_samples)
-        disk_q = self._uniform(0, 1, n_samples=n_samples)
-        disk_a = self._uniform(self.min_a_d, self.max_a_d, n_samples=n_samples)
+    @staticmethod
+    def _gamma(concentration, loc, scale, n_samples=1):
+        x = torch.distributions.Gamma(concentration, rate=1.0).sample((n_samples,))
+        return x * scale + loc
 
-        bulge_q = self._uniform(0, 1, n_samples=n_samples)
-        bulge_a = self._uniform(self.min_a_b, self.max_a_b, n_samples=n_samples)
-        return torch.stack(
-            [total_flux, disk_frac, beta_radians, disk_q, disk_a, bulge_q, bulge_a], dim=1
-        )
+
+class GalsimGalaxyDecoder:
+    def __init__(
+        self,
+        slen,
+        n_bands,
+        background,
+        pixel_scale,  # SDSS
+        psf_image_file: str,
+    ) -> None:
+
+        self.slen = slen
+        assert n_bands == 1, "Only 1 band is supported"
+        self.n_bands = 1
+        self.background = torch.zeros((self.n_bands, self.slen, self.slen), dtype=torch.float32)
+        self.background[...] = background
+        self.pixel_scale = pixel_scale
+        self.psf = load_psf_from_file(psf_image_file, self.pixel_scale)
+
+    def __call__(self, z: Tensor):
+        images = []
+        for latent in z:
+            image = self.render_galaxy(latent)["noiseless"]
```

Comment:
I see you are returnign only the noiseless images here, what do you use them for?

Reply:
This is used in the ImageDecoder when rendering an entire image. Since background and noise are handled for the entire image, we just need the noiseless galaxy here. It's a drop-in replacement for the single galaxy decoder in galaxy_net.py